src/invert/laplacexy/laplacexy.cxx

```diff
@@ -29,16 +30,49 @@ static PetscErrorCode laplacePCapply(PC pc,Vec x,Vec y) {
 }
 
 LaplaceXY::LaplaceXY(Mesh *m, Options *opt, const CELL_LOC loc)
-    : localmesh(m==nullptr ? bout::globals::mesh : m), location(loc) {
+    : localmesh(m==nullptr ? bout::globals::mesh : m), location(loc), monitor(*this) {
   Timer timer("invert");
 
   if (opt == nullptr) {
     // If no options supplied, use default
     opt = &(Options::root()["laplacexy"]);
   }
   
+  finite_volume = (*opt)["finite_volume"].doc(
+      "Use finite volume rather than finite difference discretisation."
+      ).withDefault(true);
+
+  ///////////////////////////////////////////////////
+  // Boundary condititions options
+  if (localmesh->periodicY(localmesh->xstart)) {
+    // Periodic in Y, so in the core
+    x_inner_dirichlet = (*opt)["core_bndry_dirichlet"].withDefault(false);
+  } else {
+    // Non-periodic, so in the PF region
+    x_inner_dirichlet = (*opt)["pf_bndry_dirichlet"].withDefault(true);
+  }
+  if ((*opt)["y_bndry_dirichlet"].isSet()) {
+    throw BoutException("y_bndry_dirichlet has been deprecated. Use y_bndry=dirichlet "
+                        "instead.");
```

Comment:
Nice! A simple way to deal with deprecating options, I like it.

Reply:
Doing this does skip a 'deprecation warning' stage, although would be easy enough to stick one in in a similar style, but we're allowed to break stuff in 5.0! I should probably stick in a deprecation warning into 4.4 in #1924...